Sources/Workspace/InitPackage.swift

```diff
@@ -299,6 +284,59 @@ public final class InitPackage {
                         XCTAssertEqual(\(typeName)().text, "Hello, World!")
                     }
                 
+                    static var allTests = [
+                        ("testExample", testExample),
+                    ]
+                }
+
+                """
+        }
+    }
+
+    private func writeExecutableTestsFile(_ path: AbsolutePath) throws {
+        try writePackageFile(path) { stream in
+            stream <<< """
+                import XCTest
+                import class Foundation.Bundle
+
+                final class \(moduleName)Tests: XCTestCase {
+                    func testExample() throws {
+                        // This is an example of a functional test case.
+                        // Use XCTAssert and related functions to verify your tests produce the correct
+                        // results.
+
+                        guard #available(macOS 10.13, *) else {
+                            return
+                        }
+
+                        let fooBinary = productsDirectory.appendingPathComponent("foo")
+
+                        let process = Process()
+                        process.executableURL = fooBinary
+
+                        let pipe = Pipe()
+                        process.standardOutput = pipe
+
+                        try process.run()
+                        process.waitUntilExit()
+
+                        let data = pipe.fileHandleForReading.readDataToEndOfFile()
+                        let output = String(data: data, encoding: .utf8)
+
+                        XCTAssertEqual(output, "Hello, world!\\n")
+                    }
+
+                    /// Returns path to the built products directory.
+                    var productsDirectory: URL {
+                      #if os(macOS)
+                        for bundle in Bundle.allBundles where bundle.bundlePath.hasSuffix(".xctest") {
+                            return bundle.bundleURL.deletingLastPathComponent()
+                        }
+                        fatalError()
```

Comment:
Could we add a comment to the fatalError? `fatalError("Couldn't find the products directory")`

Reply:
Why aren't we doing the same thing `SwiftPMProduct` is doing on Linux?

```
return AbsolutePath(CommandLine.arguments.first!, relativeTo: currentWorkingDirectory)
    .parentDirectory.appending(self.exec)
```

Reply:
Most likely because it wasn't implemented at the time.